runtime/major_gc.c

```diff
@@ -416,6 +417,8 @@ static void mark_slice (intnat work)
         INSTR (slice_fields += end - start;)
         INSTR (if (size > end)
                  CAML_INSTR_INT ("major/mark/slice/remain", size - end);)
+        if (size > end)
+          caml_ev_counter (EV_C_MAJOR_MARK_SLICE_REMAIN, size - end);
```

Comment:
I do not understand this code. (I realise it's adapted from the `CAML_INSTR_INT` logic above, but I don't understand that either).

This event is only posted in the `size > end` case. However, the code just below asserts that this case never occurs (either `end < size` or `end == size`).